lib/cabin/mixins/logger.rb

```diff
@@ -52,8 +52,22 @@ def level=(value)
                                   "data={}) takes at most 2 arguments")
       end
 
-      message = args[0]
-      data = args[1] || {}
+      return unless send(predicate)
+
+      if block 
+        block_results = block.call
```

Comment:
I wonder if we could simplify this as:
```
return send(level, *block.call) if block
```

This should make `logger.info { "hello" }` recursively call `logger.info("hello")`